src/app/components/UploadImage.js

```diff
@@ -36,105 +37,117 @@ const StyledUploader = styled.div`
+    values={{
+      upload_max_size: about.upload_max_size,
+      upload_extensions: about.upload_extensions,
+      upload_max_dimensions: about.upload_max_dimensions,
+      upload_min_dimensions: about.upload_min_dimensions,
+    }}
+  />
+) : (
+  <FormattedMessage
+    id="uploadImage.videoMessage"
+    defaultMessage="Drop a video file here, or click to upload a file (max size: {video_max_size}, allowed extensions: {video_extensions})"
+    values={{
+      video_max_size: about.video_max_size,
+      video_extensions: about.video_extensions,
+    }}
+  />
+);
+UploadMessage.propTypes = {
+  type: PropTypes.oneOf(['image', 'video']).isRequired,
+  about: PropTypes.shape({
+    upload_max_size: PropTypes.string.isRequired,
+    upload_extensions: PropTypes.string.isRequired,
+    upload_max_dimensions: PropTypes.string.isRequired,
+    upload_min_dimensions: PropTypes.string.isRequired,
+    video_max_size: PropTypes.string.isRequired,
+    video_extensions: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+class UploadImageComponent extends React.PureComponent {
+  onDrop = (files) => {
+    const {
+      about,
+      type,
+      onChange,
+      onError,
+    } = this.props;
     const file = files[0];
-    const { type } = this.props;
-    const extensions = type === 'image' ? this.props.about.upload_extensions : this.props.about.video_extensions;
-    const max_size = type === 'image' ? this.props.about.upload_max_size : this.props.about.video_max_size;
+    const extensions = type === 'image' ? about.upload_extensions : about.video_extensions;
+    const max_size = type === 'image' ? about.upload_max_size : about.video_max_size;
     const valid_extensions = extensions.toLowerCase().split(/[\s,]+/);
     const extension = file.name.substr(file.name.lastIndexOf('.') + 1).toLowerCase();
     if (valid_extensions.length > 0 && valid_extensions.indexOf(extension) < 0) {
-      if (this.props.onError) {
-        this.props.onError(file, this.props.intl.formatMessage(
-          messages.invalidExtension,
-          { extension, allowed_types: extensions },
-        ));
-      }
+      onError(file, <FormattedMessage
+        id="uploadImage.invalidExtension"
+        defaultMessage='The file cannot have type "{extension}". Please try with the following file types: {allowed_types}.'
+        values={{ extension, allowed_types: extensions }}
+      />);
       return;
     }
     if (file.size && unhumanizeSize(max_size) < file.size) {
-      if (this.props.onError) {
-        this.props.onError(file, this.props.intl.formatMessage(
-          messages.fileTooLarge,
-          { size: max_size },
-        ));
-      }
+      onError(file, <FormattedMessage
+        id="uploadImage.fileTooLarge"
+        defaultMessage="The file size should be less than {size}. Please try with a smaller file."
```

Comment:
need to pass {size} value here?

Reply:
Other than this, :lgtm:! Just wait to merge after deployment as per karim's comment